### PR TITLE
17 fine tuning preliminary code

### LINKED ISSUE #17: Fine-Tuning the CodeT5 Model for R Language in Bioinformatics
The CodeT5 model needs to be fine-tuned using bioinformatics-specific R datasets to enhance its code generation, debugging, and optimization capabilities. The fine-tuned model should understand Bioconductor functions commonly used in bioinformatics research.

Acceptance Criteria:
- Preprocess the collected dataset for fine-tuning.
- Fine-tune the CodeT5 model using the curated R language datasets.

### PR DESCRIPTION
**Pull request by oam67**
Fixes #17

**What was changed?**

The CodeT5 GitHub explains that if we want to fine-tune on our own dataset, we can add our own task, data path, and function to read data to the existing CodeT5 fine-tuning framework. I added a new task called "masked" to this framework. This is because we will be using masked code for our input and our output will be the portion that was masked in the input. Specifically, I have added task registration logic, data reading utilities, and file path logic to the CodeT5 fine-tuning pipeline. I also updated our .gitignore file to ignore .DS_Store since I noticed it was being pushed with the rest of my updates.

**Why was it changed?**

The default CodeT5 setup does not support fine-tuning on R code or custom task types like masked R code generation. To adapt the model for bioinformatics applications written in R, we need to enable support for a new task (masked) and custom data loading for a new dataset format. This change allows CodeT5 to be fine-tuned on our domain-specific dataset.

**How was it changed?**

I specifically updated the existing config.py to support task registration logic. I added "masked" to the --task argument's choices list. I also assigned the language r to the task. Within the existing _utils.py, I added a new data reader function called read_masked_examples. Within the existing utils.py, I registered the new task in the read_examples() function and updated file path logic in the get_file_names() function. Finally, I added .DS_Store to our .gitignore file to prevent .DS_Store from being pushed in the future.

**Screenshots that show the changes (if applicable):**
image

### CODE REVIEW

File: Fine Tuning/_utils.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
+import json
+
+
+def add_lang_by_task(target_str, task, sub_task):
```
Comment (PranaviKolouju) [inline Fine Tuning/_utils.py:4]: I know that we have defined a variety of tasks here. I am curious about how our model will work if we refine this set of tasks. Maybe this is an experiment for when we are testing, but I would be interesting in seeing if we get better performance if we fine tune on a smaller subset of tasks.

File: Fine Tuning/config.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import torch
+import logging
+import multiprocessing
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+def add_args(parser):
+    parser.add_argument("--task", type=str, required=True,
```
Comment (PranaviKolouju) [inline Fine Tuning/config.py:11]: As a general note, I think it would be helpful to refine this config file down to the bare bones of what we need for finetuning. I don't think that all parts of this code are necessary for our purposes. Maybe we can work on extracting the necessary components into our own, streamlined finetuning script.

File: Fine Tuning/utils.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        return train_fn
+    elif split == 'dev':
+        return dev_fn
+    elif split == 'test':
+        return test_fn
+    else:
+        return train_fn, dev_fn, test_fn
+
+
+def read_examples(filename, data_num, task):
```
Comment (PranaviKolouju) [inline Fine Tuning/utils.py:155]: This function is giving me something else to think about--does it expect us to have different training examples for each of these tasks? If so, there is further cleaning we will have to do before we can finetune.

**Review by AjithAkuthota23 — APPROVED** [review]
Good work!!
